pkg/gardenlet/controller/managedseed/managedseed_valueshelper.go

```diff
@@ -0,0 +1,362 @@
+	if err != nil {
+		return nil, err
+	}
+	if keyPath != nil && keyPath.(string) != "" && !strings.Contains(keyPath.(string), secrets.TemporaryDirectoryForSelfGeneratedTLSCertificatesPattern) {
+		key, err := ioutil.ReadFile(keyPath.(string))
+		if err != nil {
+			return nil, err
+		}
+		configValues, err = utils.SetToValuesMap(configValues, string(key), "server", "https", "tls", "key")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Delete server certificate and key paths
+	configValues, err = utils.DeleteFromValuesMap(configValues, "server", "https", "tls", "serverCertPath")
+	if err != nil {
+		return nil, err
+	}
+	configValues, err = utils.DeleteFromValuesMap(configValues, "server", "https", "tls", "serverKeyPath")
+	if err != nil {
+		return nil, err
+	}
+
+	return configValues, nil
+}
+
+func getParentGardenletDeployment(imageVector imagevector.ImageVector, shoot *gardencorev1beta1.Shoot) (*seedmanagementv1alpha1.GardenletDeployment, error) {
+	// Get image repository and tag
+	var imageRepository, imageTag string
+	gardenletImage, err := imageVector.FindImage("gardenlet")
+	if err != nil {
+		return nil, err
+	}
+	if gardenletImage.Tag != nil {
+		imageRepository = gardenletImage.Repository
+		imageTag = *gardenletImage.Tag
+	} else {
+		imageRepository = gardenletImage.String()
+		imageTag = version.Get().GitVersion
+	}
+
+	// Create and return result
+	return &seedmanagementv1alpha1.GardenletDeployment{
+		Image: &seedmanagementv1alpha1.Image{
+			Repository: &imageRepository,
+			Tag:        &imageTag,
+		},
+		PodAnnotations: getParentPodAnnotations(shoot),
+	}, nil
+}
+
+func getParentImageVectorOverwrite() (string, error) {
+	var imageVectorOverwrite string
+	if overWritePath := os.Getenv(imagevector.OverrideEnv); len(overWritePath) > 0 {
+		data, err := ioutil.ReadFile(overWritePath)
+		if err != nil {
+			return "", err
+		}
+		imageVectorOverwrite = string(data)
+	}
+	return imageVectorOverwrite, nil
+}
+
+func getParentComponentImageVectorOverwrites() (string, error) {
+	var componentImageVectorOverwrites string
+	if overWritePath := os.Getenv(imagevector.ComponentOverrideEnv); len(overWritePath) > 0 {
+		data, err := ioutil.ReadFile(overWritePath)
+		if err != nil {
+			return "", err
+		}
+		componentImageVectorOverwrites = string(data)
+	}
+	return componentImageVectorOverwrites, nil
+}
+
+var minimumAPIServerSNISidecarConstraint *semver.Constraints
+
+func init() {
+	var err error
+	// 1.13.0-0 must be used or no 1.13.0-dev version can be matched
+	minimumAPIServerSNISidecarConstraint, err = semver.NewConstraint(">= 1.13.0-0")
+	utilruntime.Must(err)
+}
+
+func getParentPodAnnotations(shoot *gardencorev1beta1.Shoot) map[string]string {
+	// If APIServerSNI is enabled for the seed cluster then the gardenlet must be restarted, so the Pod injector would
+	// add `KUBERNETES_SERVICE_HOST` environment variable.
+	if gardenletfeatures.FeatureGate.Enabled(features.APIServerSNI) {
```

Comment:
See above, I don't see why we should do this. My assumption with introducing `MergeWithParent=false` (which was clearly not there before at all) was that we could in some cases find it beneficial to deploy a gardenlet that was configured in a way completely different from the parent, with all the resulting consequences. While as far as I understand this is not going to be used a lot at the moment (perhaps not used at all), it might become more useful in the future once we introduce the `ManagedSeedSet` for managing multiple managed seeds based on a template.

If we don't find `MergeWithParent=false" useful, perhaps we should better remove it rather than introducing a strange hybrid that only uses certain parent values, but not others.

Reply:
After clarifying what `MergeWithParent` means I agree this pod annotation should not depend on it, see above.

Reply:
Done